telemetry/telemetry.go

```diff
@@ -440,9 +430,10 @@ func (reportMgr *ReportManager) ReportToBytes() (report []byte, err error) {
 		err = fmt.Errorf("[Telemetry] Invalid report type")
 	}
 
-	if err == nil {
```

Comment:
why did we change this? It's the same behaviour right?

Reply:
i checked this and it didn't work
> err= fmt.Errorf("[Telemetry] Invalid report type")

outside the switch case its showing nil